### PR TITLE
GRR Yara hunt

### PR DESCRIPTION
**Pull request by tomchop**
New module that launches hunts based on Yara rules.

### CODE REVIEW

File: dftimewolf/lib/collectors/grr_base.py [hunk 1]
```diff
@@ -137,9 +137,15 @@ def _WrapGRRRequestWithApproval(
         approval = grr_object.CreateApproval(
             reason=self.reason, notified_users=self.approvers)
         approval_sent = True
-        approval_url = ('{0:s}/v2/clients/{1:s}/users/{2:s}/approvals/{3:s}'.
-                        format(self.grr_url, approval.client_id,
-                               approval.username,
-                               approval.approval_id))
+        if hasattr(approval, 'client_id'):
+          approval_url = ('{0:s}/v2/clients/{1:s}/users/{2:s}/approvals/{3:s}'.
+                          format(self.grr_url, approval.client_id,
```
Comment (jleaniz) [inline dftimewolf/lib/collectors/grr_base.py:142]: nit: should we use f-strings here and throughout? Other modules use f-strings.

File: dftimewolf/lib/collectors/grr_hunt.py [hunk 2]
```diff
@@ -45,51 +44,69 @@ def __init__(self,
     self.match_mode = ""
     self.client_operating_systems : Set[str] = set()
     self.client_labels : List[str] = []
+    self.hunt_runner_args = None  # type: Optional[grr_flows.HuntRunnerArgs]
+    self.hunt_description = ''
```
Comment (jleaniz) [inline dftimewolf/lib/collectors/grr_hunt.py:48]: ```self.match_mode = ""``` -> ``` self.match_mode: str = ""``` ?

Comment (jleaniz) [inline dftimewolf/lib/collectors/grr_hunt.py:48]: ```self.hunt_description = ''``` -> ```self.hunt_description: str = ""``` ?

File: dftimewolf/lib/collectors/grr_hunt.py [hunk 3]
```diff
@@ -45,51 +44,69 @@ def __init__(self,
...
 
   def HuntSetup(
       self,
       match_mode: Optional[str],
       client_operating_systems: Optional[str],
-      client_labels: Optional[str]) -> None:
+      client_labels: Optional[str],
+      extra_hunt_runner_args) -> None:
```
Comment (jleaniz) [inline dftimewolf/lib/collectors/grr_hunt.py:55]: extra_hunt_runner_args type?

File: dftimewolf/lib/collectors/grr_hunt.py [hunk 4]
```diff
@@ -45,51 +44,69 @@ def __init__(self,
...
     Args:
-      match_mode (Optional[str]): match mode of the client rule set (ALL or
-          ANY).
-      client_operating_systems (Optional[str]): a comma separated list of client
-          OS types (win, osx or linux).
-      client_labels (Optional[str]): a comma separated list of client labels.
+      match_mode: match mode of the client rule set (ALL or ANY).
+      client_operating_systems: a comma separated list of client OS types (win,
+          osx or linux).
+      client_labels: a comma separated list of client labels.
```
Comment (jleaniz) [inline dftimewolf/lib/collectors/grr_hunt.py:62]: nit: Do we need to add ``` extra_hunt_runner_args``` to this docstring?

File: dftimewolf/lib/collectors/grr_hunt.py [hunk 5]
```diff
@@ -411,7 +399,129 @@ def Process(self) -> None:
...
+      client_limit: The number of clients to run the hunt on. 0 for no limit.
+      process_ignorelist: A list of process executable paths to ignore.
+    """
+    self.GrrSetUp(
+        reason, grr_server_url, grr_username, grr_password, approvers=approvers,
+        verify=verify, message_callback=self.PublishMessage)
+
+    if isinstance(process_ignorelist, list):
+      joined = "|".join(process_ignorelist)
+    elif isinstance(process_ignorelist, str):
```
Comment (jleaniz) [inline dftimewolf/lib/collectors/grr_hunt.py:475]: It's confusing to have this if we're saying that process_ignorelist is a List[str]. I guess a client of this class could just pass a string but that's not intended.  I would probably throw an error here or adjust the type to ```List[str] | str```

File: tests/lib/collectors/grr_hunt.py [hunk 6]
```diff
@@ -335,5 +333,90 @@ def testGetAndWriteWrongResults(self, mock_list_results):
...
+      flow_name='YaraProcessScan',
+      flow_args=flows_pb2.YaraProcessScanRequest(
+          yara_signature=('import "pe"\nimport "math"\n\nrule test { '
+                          'strings: $a = "abcdefg" condition: $a and pe.DLL }'
+                          '\n\nrule test { strings: $a = "0123456" condition:'
+                          ' $a and math.entropy($a) }'),
+          ignore_grr_process=True,
+          process_regex=r"(?i)^(?!\.exe|ignore1).*",
+          dump_process_on_match=True,
+          process_dump_size_limit= 268_435_456,
```
Comment (jleaniz) [inline tests/lib/collectors/grr_hunt.py:415]: what is this 268_435_456? is it mean tot be an int?

**Review by jleaniz — CHANGES_REQUESTED** [review]
LGTM, left some comments / suggestions.

File: tests/lib/collectors/grr_hunt.py (hunk 6 above)
Comment (tomchop) [reply tests/lib/collectors/grr_hunt.py:415]: `_` are a way to separate thousands in integers. so it's equivalent to `268435456` but just easier to read (supposedly)

**Pull request merged**